addons/hr_expense/models/hr_expense.py

```diff
@@ -481,15 +481,17 @@ def message_new(self, msg_dict, custom_values=None):
             except ValueError:
                 price = 1.0
 
+        company_id = employee.company_id.id
         custom_values.update({
             'name': expense_description.strip(),
             'employee_id': employee.id,
             'product_id': product.id,
             'product_uom_id': product.uom_id.id,
-            'tax_ids': [(4, tax.id, False) for tax in product.supplier_taxes_id],
+            # If company_id is set, always filter taxes by the company
+            'tax_ids': [(4, tax.id, False) for tax in product.supplier_taxes_id.filtered(lambda r: not company_id or r.company_id.id == company_id)],
```

Comment:
`'tax_ids': [(4, tax.id) for tax in product.supplier_taxes_id.filtered(lambda x: x.company_id == company)],`